gms/factories/src/main/java/com/linkedin/gms/factory/common/Neo4jDriverFactory.java

```diff
@@ -27,19 +27,23 @@
   @Value("${NEO4J_MAX_CONNECTION_ACQUISITION_TIMEOUT_IN_SECONDS:60}")
   private Long neo4jMaxConnectionAcquisitionTimeout;
 
-  @Value("${NEO4j_MAX_CONNECTION_LIFETIME_IN_HOURS:1}")
+  @Value("${NEO4j_MAX_CONNECTION_LIFETIME_IN_SECONDS:3600}")
   private Long neo4jMaxConnectionLifetime;
 
   @Value("${NEO4J_MAX_TRANSACTION_RETRY_TIME_IN_SECONDS:30}")
   private Long neo4jMaxTransactionRetryTime;
 
+  @Value("${NEO4J_CONNECTION_LIVENESS_CHECK_TIMEOUT_IN_SECONDS:-1}")
+  private Long neo4jConnectionLivenessCheckTimeout;
```

Comment:
I wonder if it would benefit other folks to give this a reasonable default- I imagine if you're running into liveness issues others may be as well. What value have you found the be successful here?

Reply:
0 has worked well under testing conditions, however it stresses Neo4j more. From the [docs](https://neo4j.com/docs/api/java-driver/current/org/neo4j/driver/Config.ConfigBuilder.html#withConnectionLivenessCheckTimeout-long-java.util.concurrent.TimeUnit-):

> If this option is set too low, an additional network call will be incurred when acquiring a connection, which causes a performance hit.

So I thought, better to have it as the default in the java-driver, i.e disabled, and only tune it when needed. 
Wdyt?

Reply:
I agree 0 is a bit aggressive as a default- where you able to find success with a slightly higher timeout? Checking every 5 minutes, for example, seems like a reasonable default. If that doesn't work in your case, I agree `-1` seems like a safer default for now until we can get more data points.

Reply:
> Checking every 5 minutes, for example, seems like a reasonable default.

Just to clarify, this is not a keep-alive feature that one would expect, but rather a feature that verifies that the connection is alive upon use. 
So not as useful as `NEO4j_MAX_CONNECTION_LIFETIME_IN_SECONDS` perhaps, but it still adds value, when the connection to Neo4j is flaky or as an extra check to avoid exceptions in the GUI.

If this setting is used and the connection is closed, it will still generate an exception on the GMS, but the connection will be refreshed and the frontend won't notice anything.

So, yes, I would prefer if -1 is kept as default, since it's more of a power-user setting, other than perhaps a delay.